src/SFA.DAS.EmployerFinance.MessageHandlers.UnitTests/EventHandlers/ExpireFundsCommandHandlerTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using NServiceBus;
+using NUnit.Framework;
+using SFA.DAS.EmployerFinance.Data;
+using SFA.DAS.EmployerFinance.MessageHandlers.CommandHandlers;
+using SFA.DAS.EmployerFinance.Messages.Commands;
+using SFA.DAS.EmployerFinance.Models.ExpiredFunds;
+using SFA.DAS.EmployerFinance.Models.Levy;
+using SFA.DAS.EmployerFinance.Models.Payments;
+using SFA.DAS.EmployerFinance.Types.Models;
+using SFA.DAS.Testing;
+
+namespace SFA.DAS.EmployerFinance.MessageHandlers.UnitTests.EventHandlers
+{
+    [TestFixture]
+    public class ExpireAccountFundsCommandHandlerTests : FluentTest<ExpireAccountFundsCommandHandlerTestsFixture>
+    {
+        [Test]
+        public Task Handle_WhenHandlingExpireAccountFundsCommand_ThenShouldGetTheFundsIn()
+        {
+            return RunAsync(f => f.Handle(),
+                f => f.MockLevyFundsInRepository.Verify(r => r.GetLevyFundsIn(f.ExpectedAccountId)));
+        }
+
+        [Test]
+        public Task Handle_WhenHandlingExpireAccountFundsCommand_ThenShouldGetTheFundsOut()
+        {
+            return RunAsync(f => f.Handle(),
+                f => f.MockPaymentFundsOutRepository.Verify(r => r.GetPaymentFundsOut(f.ExpectedAccountId)));
+        }
+
+        [Test]
+        public Task Handle_WhenHandlingExpireAccountFundsCommand_ThenShouldGetTheExistingExpiredFunds()
+        {
+            return RunAsync(f => f.Handle(), f => f.MockExpiredFundsRepository.Verify(x => x.Get(f.ExpectedAccountId)));
+        }
+
+        [Test]
+        public Task Handle_WhenHandlingExpireAccountFundsCommand_ThenShouldCallTheExpiredFundsAlgorithm()
+        {
+            return RunAsync(f => f.Handle(), f => f.MockExpiredFunds.Verify(x => x.GetExpiringFunds(
+                It.Is<Dictionary<CalendarPeriod, decimal>>(fi => ExpiredFundsComparisonHelper.AreFundsInEqual(f.FundsIn, fi)),
```

Comment:
The `ExpiredFundsComparisonHelper` calls could be methods on your `ExpireAccountFundsCommandHandlerTestsFixture` instead if you prefer e.g.

```c#
It.Is<Dictionary<CalendarPeriod, decimal>>(fi => f.AreFundsInEqual(f.FundsIn, fi))
```